src/core/execute.c

```diff
@@ -2385,7 +2385,10 @@ static int exec_child(
 
         exec_context_tty_reset(context, params);
 
-        if (!manager_is_confirm_spawn_disabled(unit->manager)) {
+        /* For some reasons units remaining in the same process group
+         * as PID 1 fail to acquire the console even if it's not used
+         * by any process. So skip the confirmation question for them. */
+        if (!context->same_pgrp && !manager_is_confirm_spawn_disabled(unit->manager)) {
```

Comment:
Well manager_is_confirm_spawn_disabled() is testing that the confirmation stuff is disabled *globaly* not at the unit level, so I'm really not sure.

However introducing unit_shall_confirm_spawn() which tests both at the unit level and globally might be nicer indeed.